app/helpers/foreman_host_reports/reports_dashboard_helper.rb

```diff
@@ -0,0 +1,72 @@
+module ForemanHostReports
+  module ReportsDashboardHelper
+    def host_reports_searchable_links(name, search, counter, format)
+      content_tag :li, :style => "margin-bottom: 5px" do
+        content_tag(:span, sanitize('&nbsp;'), :class => 'label', :style => "background-color: #{host_reports_report_color[counter]}") +
+          sanitize('&nbsp;') +
+          link_to(name, hosts_path(:search => search), :class => "dashboard-links") +
+          content_tag(:span, send(counter, format), class: 'pull-right')
+      end
+    end
+
+    def host_reports_report_color
+      {
+        :change_hosts => "#4572A7",
+        :failure_hosts => "#AA4643",
+        :nochange_hosts => "#DB843D",
+        :disabled_hosts => "#92A8CD",
+      }
+    end
+
+    def host_reports_get_overview(options = {})
+      format = options[:format]
+      state_labels = {
+        change_hosts: _('Change'),
+        failure_hosts: _('Failure'),
+        nochange_hosts: _('No Change'),
+        disabled_hosts: _('Disabled alerts'),
+      }
+
+      {
+        data: state_labels.map { |key, label| [label, send(key, format), host_reports_report_color[key]] },
+        searchUrl: hosts_path(search: '~VAL~'),
+        searchFilters: state_labels.each_with_object({}) do |(key, filter), filters|
+          filters[filter] = send(key, format)
+        end,
+      }
+    end
+
+    def latest_reports
+      HostReport.authorized(:view_host_reports).my_reports
+                .reorder(:last_report => :desc)
+                .limit(9)
+                .preload(:host)
```

Comment:
This query does not work and it errors out. It renders to the following SQL: `"SELECT \"host_reports\".* FROM \"host_reports\" ORDER BY \"last_report\" DESC LIMIT 9"` (under admin account so it is more simple). The reason is that last_report must be changed to reported_at:

    HostReport.authorized(:view_host_reports).my_reports.reorder(:reported_at => :desc).limit(9).preload(:host)

I am not sure how this could work for you, last report widgets could never display anything but the SQL error.

Reply:
Also please add a test case for this method, it is untested at the moment.